### PR TITLE
[TOSA] Expand Torch to TOSA legalization coverage

### PR DESCRIPTION
**Pull request by justin-ngo-arm**
- Add/Extend Torch to TOSA legalization for the following ops:
  + Add aten.threshold_backward
  + Fix aten.threshold
  + Re-implement aten.broadcast_to using tosa.reshape and tosa.tile
  + Add support for rank 0 index for aten.index_select
  + Fix aten.index_put.hacked_twin
  + Add aten.uniform
  + Add aten.logical_and
- Update xfail_sets.py with new e2e results
- Add LIT tests to basic.mlir for newly added ops

Change-Id: I8910564a049d18293284fe2e55e82bc1d2cf10e3

### CODE REVIEW

**Review by sjarus — APPROVED** [review]
Looks reasonable. Adding @sahas3 for potential feedback around tensor.hacked_twin.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp [hunk 1]
```diff
@@ -6632,6 +6622,140 @@ LogicalResult ConvertAtenOp<AtenDiagEmbedOp>::matchAndRewrite(
   return success();
 }
 
+// Legalization for aten.uniform
+// Since TOSA hasn't got a built-in random generator yet, we will use
+// std::uniform_real_distribution with the std::default_random_engine from C++
+// <random> library
+template <>
+LogicalResult ConvertAtenOp<AtenUniformOp>::matchAndRewrite(
```
Comment (sahas3) [inline lib/Conversion/TorchToTosa/TorchToTosa.cpp:6630]: Does this need a LIT test too or since it's covered by e2e tests the LIT one can be skipped?

**Review by sahas3 — APPROVED** [review]
LGTM, just one clarification question.

File: lib/Conversion/TorchToTosa/TorchToTosa.cpp (hunk 1 above)
Comment (justin-ngo-arm) [reply lib/Conversion/TorchToTosa/TorchToTosa.cpp:6630]: Thank you for the comment, I will add a LIT test for that op too.

**Pull request merged**